device/common/include/traccc/seeding/device/impl/count_triplets.ipp

```diff
@@ -0,0 +1,164 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021-2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+#pragma once
+
+// Project include(s).
+#include "traccc/seeding/doublet_finding_helper.hpp"
+#include "traccc/seeding/triplet_finding_helper.hpp"
+
+// VecMem include(s).
+#include <vecmem/memory/device_atomic_ref.hpp>
+
+namespace traccc::device {
+
+TRACCC_HOST_DEVICE
+void count_triplets(
+    const std::size_t globalIndex, const seedfinder_config& config,
+    const sp_grid_const_view& sp_view,
+    const doublet_counter_container_types::const_view doublet_counter_view,
+    const vecmem::data::vector_view<const prefix_sum_element_t>&
+        doublet_ps_view,
+    const doublet_container_view mid_bot_doublet_view,
+    const doublet_container_view mid_top_doublet_view,
+    triplet_counter_container_types::view triplet_view) {
+
+    // Create device copy of input parameters
+    vecmem::device_vector<const prefix_sum_element_t> doublet_prefix_sum(
+        doublet_ps_view);
+
+    // Check if anything needs to be done.
+    if (globalIndex >= doublet_prefix_sum.size()) {
+        return;
+    }
+
+    // Create device copy of input parameters
+    const device_doublet_container mid_bot_doublet_device(mid_bot_doublet_view);
+
+    // Get ids
+    const prefix_sum_element_t ps_idx = doublet_prefix_sum[globalIndex];
+    const unsigned int bin_idx = ps_idx.first;
+    const unsigned int item_idx = ps_idx.second;
+
+    // Get current mid bottom doublet
+    const doublet mid_bot =
+        mid_bot_doublet_device.get_items().at(bin_idx).at(item_idx);
+
+    // Create device copy of input parameters
+    const device::doublet_counter_container_types::const_device
+        doublet_counter_device(doublet_counter_view);
+    const device_doublet_container mid_top_doublet_device(mid_top_doublet_view);
+
+    // Create device copy of output parameter
+    triplet_counter_container_types::device triplet_counter(triplet_view);
+
+    // Get all spacepoints
+    const const_sp_grid_device internal_sp_device(sp_view);
+
+    // Get internal spacepoints for current bin
+    const unsigned int& num_compat_spM_per_bin =
```

Comment:
Instead of using a reference here, let's just get this variable into a "const value". So that the value can be handled in a register, and the code would not think about loading it from global memory on every access. (I.e. just remove the `&` from the declaration.)